Allows phpstormhelper to handle and account for coverage options

billypoke opened the pull request:
Closes #685

billypoke committed:
Allows phpstormhelper to handle and account for coverage options

Slamdunk commented on test/Unit/Util/PhpstormHelperTest.php:
```diff
@@ -94,5 +94,76 @@ public function providePhpstormCases(): Generator
+        $argv[] = $phpunitBinary;
+        $argv[] = '--runner';
+        $argv[] = 'WrapperRunner';
+        $argv[] = '-dpcov.enabled=1';
```
This isn't the order I get in my PHPStorm:

```php
array (
  0 => './bin/paratest_for_phpstorm',
  1 => '-dpcov.enabled=1',
  2 => '/home/tessarotto/repos/paratest/vendor/phpunit/phpunit/phpunit',
  3 => '--runner',
  4 => 'WrapperRunner',
  5 => '--coverage-clover',
  6 => '/home/tessarotto/.cache/JetBrains/PhpStorm2022.2/coverage/paratest@ParaTest.xml',
  7 => '--configuration',
  8 => '/home/tessarotto/repos/paratest/phpunit.xml',
  9 => '--teamcity',
)
```

Can you add this to data sets?
At this point the original `unset($argv[1]);` is not enough, we need to cycle all the `$argv` var and remove the first entry that ends with `/phpunit`

billypoke replied:
Ok, the test cases are updated to match, the order of arguments on my machine matches yours. I also switched to searching for the necessary argv keys to unset instead of hard-coding the array keys. Let me know if this is what you were aiming for :)

Slamdunk reviewed (changes requested):
Hi, thank you for this proposal.

I feel that this doesn't solve the underlying issue though. PhpStorm adds `-dpcov.enabled=1` to be sure PCOV is enabled, that indeed might be disabled in the system configuration.
Removing the flag here is necessary, but PCOV or xDebug are still disabled in such cases.
We should edit the `README.md` adding that:

1. It is suggested to already have PCOV/xDebug enabled for code coverage
2. When it's not a viable, add a second Configuration with `-dpcov.enabled=1`/`'-dxdebug.mode=coverage'` prepended to the `./vendor/bin/paratest_for_phpstorm` interpreter option

billypoke committed:
Use search functions instead of assuming position of arguments
Put test case arguments in expected order
Directly require symfony polyfill for php8.0 function

billypoke committed:
Update README.md to indicate additional configuration needed for coverage with PHPStorm

billypoke committed:
Sort composer reqs, add config option to auto-sort in the future

Slamdunk committed:
Add compatibility with PHP 7.3

Slamdunk committed:
Ensure test name is a string

Slamdunk committed:
Merge branch '6.x' into phpstorm_runner_coverage
